app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -643,6 +691,33 @@ class HomeFragment : Fragment() {
         navigateToSearch()
     }
 
+    private fun handleToolbarPasteAndGo(url: String) {
+        val context = requireContext()
+
+        (context as HomeActivity).openToBrowserAndLoad(
+            searchTermOrURL = url,
+            newTab = true,
+            from = BrowserDirection.FromHome,
+            engine = context.components.search.provider.getDefaultEngine(context)
+        )
+
+        val event = if (url.isUrl()) {
+            Event.EnteredUrl(false)
+        } else {
+            val searchAccessPoint = ACTION
```

Comment:
In SearchController we use ` activity.settings().incrementActiveSearchCount()` on this else statement. This probably is applicable here as well

Reply:
Can't help but wonder if we could reuse the logic from SearchController/BrowserToolbarView since everything is quite similar